edx_proctoring/api.py

```diff
@@ -1052,9 +1059,26 @@ def create_exam_attempt(exam_id, user_id, taking_as_proctored=False):
         )
         raise StudentExamAttemptOnPastDueProctoredExam(err_msg)
 
+    name_affirmation_service = get_runtime_service('name_affirmation')
+    should_create_verified_name = False
+
     if taking_as_proctored:
-        external_id, force_status = _register_proctored_exam_attempt(
-            user_id, exam_id, exam, attempt_code, review_policy
+        verified_name = None
+        if name_affirmation_service and is_verified_name_enabled:
```

Comment:
Can we move this block of logic into a separate function? Private or not. the logic looks cleaner and becomes more re-usable when it can have a consistent input and output